test/e2e_env/kubernetes/gateway/delegated_meshcircuitbreaker.go

```diff
@@ -18,13 +18,12 @@ func CircuitBreaker(config *delegatedE2EConfig) func() {
 
 	return func() {
 		BeforeAll(func() {
-			Expect(DeleteMeshResources(
-				kubernetes.Cluster,
-				config.mesh,
-				core_mesh.CircuitBreakerResourceTypeDescriptor,
-				core_mesh.RetryResourceTypeDescriptor,
-				v1alpha1.MeshCircuitBreakerResourceTypeDescriptor,
-			)).To(Succeed())
+			Eventually(func() error {
```

Comment:
I've noticed that sometimes `DeleteMeshResources` silently succeeds before the default resource is created. Change this to remove with an error, so when the policy doesn't exist yet, we are going to retry.

Reply:
Instead of putting it in `Eventually` I would modify `DeleteMeshPolicyOrError` and use `github.com/gruntwork-io/terratest/modules/retry`

```go
func DeleteMeshPolicyOrError(
	cluster Cluster,
	descriptor core_model.ResourceTypeDescriptor,
	policyName string,
) error {
	_, err := retry.DoWithRetryE(
		cluster.GetTesting(),
		"delete policy",
		10,
		time.Second,
		func() (string, error) {
			return k8s.RunKubectlAndGetOutputE(
				cluster.GetTesting(),
				cluster.GetKubectlOptions(Config.KumaNamespace),
				"delete",
				descriptor.KumactlArg,
				policyName,
			)
		},
	)
	
	return err
}
```
and then just
```go
Expect(DeleteMeshPolicyOrError(
	kubernetes.Cluster,
	meshretry_api.MeshRetryResourceTypeDescriptor,
	fmt.Sprintf("mesh-retry-all-%s", config.mesh),
).To(Succeed())
```